src/server/data/kv-redis.ts

```diff
@@ -0,0 +1,53 @@
+import { createClient } from "@node-redis/client"
+import { ConfigParam } from "../config-param"
+import { KVIterator, KVProvider } from "./kv-provider"
+
+export type RedisClientConfig = {
+    url: string
+}
+
+export function redisClientConfigParams(prefix: string) {
+    return {
+        url: ConfigParam.string(
+            `${prefix}-redis-client-url`,
+            "redis://localhost:6379",
+            `URL to ${prefix} redis`,
+        ),
+    }
+}
+
+export function redisProvider(config: RedisClientConfig): KVProvider {
+    const client = createClient({
+        url: config.url,
+    })
+    let connected = false
+    async function ensureConnected() {
+        if (!connected) {
+            await client.connect()
+            connected = true
+        }
+    }
+    return {
+        async get<T>(key: string): Promise<T | null | undefined> {
+            await ensureConnected()
+            const value = await client.get(key)
```

Comment:
The type of the entry in Redis will be "List", and `get` will not work. Instead of `get(key)` we should use `lRange(key, 0, -1-skipCount)`. I expect that it will return the array of elements sorted from latest to earliest:

```
127.0.0.1:6379> LRANGE test 0 -1
(empty array)
127.0.0.1:6379> LPUSH test 0 1 2
(integer) 3
127.0.0.1:6379> LPUSH test 3 4 5
(integer) 6
127.0.0.1:6379> LRANGE test 0 -1
1) "5"
2) "4"
3) "3"
4) "2"
5) "1"
6) "0"
127.0.0.1:6379> LRANGE test 0 -4
1) "5"
2) "4"
3) "3"
127.0.0.1:6379> GET test
(error) WRONGTYPE Operation against a key holding the wrong kind of value
127.0.0.1:6379>
```